doc/markdownlinkconverter/markdownlinkconverter.c

```diff
@@ -0,0 +1,452 @@
+	}
+	FILE * output = fopen (TEMP_FILENAME, "w+");
+	if (!output)
+	{
+		fclose (input);
+		fprintf (stderr ,"fopen Error\n");
+		return EXIT_FAILURE;
+	}
+
+	//save start of each file
+	fpos_t startInput;
+	fpos_t startTempFile;
+	if (fgetpos (input, &startInput))
+		exitError (input, output, "fgetpos");
+	if (fgetpos (output, &startTempFile))
+		exitError (input, output, "fgetpos");
+
+	if(!convertTitle (input, output, filenameInElektra))
+	{
+		/* No title found in file, therefore generate one and
+		 * print it out.
+		 */
+		//reset input file
+		if (fsetpos (input, &startInput))
+			exitError (input, output, "fsetpos");
+
+		fclose (output);
+		output = stdout;
+		//Generate Title
+		char * title = strrchr (filename, FOLDER_DELIMITER);
+		if (title == NULL)
+			exitError (input, NULL, "parsed file path invalid");
+
+		// print Title + Header
+		fprintf (output, "# %s # {#", &title[1]);
+		printConvertedPath (output, filenameInElektra);
+		fprintf (output, "}\n");
+		//2nd pass
+		convertLinks (input, output, filenameInElektra);
+		fclose (input);
+	} else
+	{
+		fclose (input);
+		//reset temp file
+		if (fsetpos (output, &startTempFile))
+			exitError (output, NULL, "fsetpos");
+
+		//2nd pass
+		convertLinks (output, stdout, filenameInElektra);
+		fclose (output);
+	}
+	remove (TEMP_FILENAME);
+
+	return EXIT_SUCCESS;
+}
+
+void printTarget(FILE * output, char * target, char * filenameInElektra)
+{
+	fprintf (output, "@ref ");
+	// distinguish between relative and absolute targets
+	if (target[0] != '/')
+	{
+		//copy filenameInElektra and extend it with FOLDER_DELIMITER at start.
+		char copyFilenameData[strlen (filenameInElektra) + 2];
+		strcpy (&copyFilenameData[1], filenameInElektra);
+		copyFilenameData[0] = FOLDER_DELIMITER;
+		char * copyFilename = copyFilenameData;
+		// determine how many times "../" is used in the target
+		int folderBack = 0;
+		while (!strncmp ("../", target, 3))
+		{
+			++folderBack;
+			target = target + 3;
+		}
+		// go folderBack times upwards in filenameInElektra
+		char * lastFolderDelimiter = strrchr (&copyFilenameData[1], FOLDER_DELIMITER);
+		if (lastFolderDelimiter != NULL)
+		{
+			*lastFolderDelimiter = '\0';
+			// not in root so target needs to be extended with path
+			while (folderBack>0)
+			{
+				--folderBack;
+				lastFolderDelimiter = strrchr (copyFilename, FOLDER_DELIMITER);
+				if (lastFolderDelimiter == NULL)
+				{
+					fprintf (output, "INVALID LINK");
+					return;
+				}
+				*lastFolderDelimiter = '\0';
+			}
+			// if copyFilename was not reduced print it
+			if (copyFilename[0] != '\0')
+			{
+				printConvertedPath (output, &copyFilename[1]);
+				fprintf (output, "_");
+			}
+		}
+	}else
+	{
+		++target; // remove starting /
+	}
+	printConvertedPath (output, target);
+}
+
+/* This helper converts and prints a given path, all slashes or backslashes
+ * and points will be replaced by underscore.
+ */
+void printConvertedPath (FILE * output, char * path)
```

Comment:
yes that is right, even more functions can be defined as static, but I think there will be no files which use the markdownlinkconverter